tests/terraform/createcluster_test.go

```diff
@@ -0,0 +1,231 @@
+package terraform
+
+import (
+	"flag"
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var tfVars = flag.String("tfvars", "/tests/terraform/modules/config/local.tfvars", "custom .tfvars file from base project path")
+var destroy = flag.Bool("destroy", false, "a bool")
+
+var failed = false
+
+func Test_TFClusterCreateValidation(t *testing.T) {
+	RegisterFailHandler(Fail)
+	flag.Parse()
+
+	RunSpecs(t, "Create Cluster Test Suite")
+}
+
+var _ = Describe("Test:", func() {
+	Context("Build Cluster:", func() {
+		It("Starts up with no issues", func() {
+			status, err := BuildCluster(&testing.T{}, *tfVars, false)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(status).To(Equal("cluster created"))
+			defer GinkgoRecover()
+			fmt.Println("Server Node IPS:", MasterIPs)
+			fmt.Println("Agent Node IPS:", WorkerIPs)
+			PrintFileContents(KubeConfigFile)
+			Expect(MasterIPs).ShouldNot(BeEmpty())
+			if NumWorkers > 0 {
+				Expect(WorkerIPs).ShouldNot(BeEmpty())
+			} else {
+				Expect(WorkerIPs).Should(BeEmpty())
+			}
+			Expect(KubeConfigFile).ShouldNot(BeEmpty())
+		})
+
+		It("Checks Node and Pod Status", func() {
+			defer func() {
+				_, err := GetNodes(KubeConfigFile, true)
+				if err != nil {
+					fmt.Println("Error retrieving nodes: ", err)
+				}
+				_, err = GetPods(KubeConfigFile, true)
+				if err != nil {
+					fmt.Println("Error retrieving pods: ", err)
```

Comment:
Should we exit here if pods can't be gotten?

Reply:
My goal is for this to print the pods at the very end of this test. The test will fail in the block below:
```

			Eventually(func(g Gomega) {
				pods, err := GetPods(kubeConfigFile, false)
				g.Expect(err).NotTo(HaveOccurred())
				for _, pod := range pods {
					if strings.Contains(pod.Name, "helm-install") {
						g.Expect(pod.Status).Should(Equal("Completed"), pod.Name)
					} else {
						g.Expect(pod.Status).Should(Equal("Running"), pod.Name)
						g.Expect(pod.Restarts).Should(Equal("0"), pod.Name)
						numRunning := re.FindAllString(pod.Ready, 2)
						g.Expect(numRunning[0]).Should(Equal(numRunning[1]), pod.Name, "should have all containers running")
					}
				}
			}, "600s", "5s").Should(Succeed())
```
but this specific bit on lines 51-53 is more to just print for visibility so we see what is (or isn't) running at the end of this test.